File: BackEnd.v2.1/demo/src/main/java/com/example/demo/models/DevisRequest.java
package com.example.demo.models;
import com.example.demo.embedded.PotentialCustomer;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIdentityReference;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.Data;

import java.util.UUID;

@Entity
@Data
@Table(name = "devisRequest")
public class DevisRequest {
    @Id
    @GeneratedValue
    private UUID id;
    private String firstName;
    private String lastName;
    private String email;
    private String phone;
    private String location;
    private String post_code;
    private String building_type;
    private Integer available_area;
    private String roof_type;
    private Boolean electricity_access;
    private Float consumption;
    private String status;
    @OneToOne(mappedBy = "devisRequest")
    private Devis devis;
    @JsonIdentityInfo(generator= ObjectIdGenerators.PropertyGenerator.class, property="id")
    @JsonIdentityReference(alwaysAsId=true)
    @ManyToOne
    private Company company;
}

File: FrontEnd.v2/src/app/models/DevisRequest.ts
import {Company} from "./Company";

export interface DevisRequest {
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  available_area: number;
  building_type: string;
  companyDto: Company;
  consumption: number;
  electricity_access: boolean;
  id?: any;
  location: string;
  post_code: string;
  roof_type: string;
  status: string;
}
